app/code/Magento/AmqpStore/Test/Unit/Plugin/AsynchronousOperations/MassConsumerEnvelopeCallbackTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AmqpStore\Test\Unit\Plugin\AsynchronousOperations;
+
+use Magento\AmqpStore\Plugin\AsynchronousOperations\MassConsumerEnvelopeCallback;
+use Magento\AsynchronousOperations\Model\MassConsumerEnvelopeCallback as SubjectMassConsumerEnvelopeCallback;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\MessageQueue\EnvelopeFactory;
+use Magento\Framework\MessageQueue\EnvelopeInterface;
+use Magento\Framework\MessageQueue\QueueInterface;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Store\Model\Store;
+use Magento\Store\Model\StoreManagerInterface;
+use PhpAmqpLib\Wire\AMQPTable;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class MassConsumerEnvelopeCallbackTest extends TestCase
+{
+    /**
+     * @var MassConsumerEnvelopeCallback
+     */
+    private $massConsumerEnvelopeCallbackPlugin;
+
+    /**
+     * @var EnvelopeFactory|MockObject
+     */
+    private $envelopeFactoryMock;
+
+    /**
+     * @var StoreManagerInterface|MockObject
+     */
+    private $storeManagerMock;
+
+    /**
+     * @var \Psr\Log\LoggerInterface|MockObject
+     */
+    private $loggerMock;
+
+    /**
+     * @var SubjectMassConsumerEnvelopeCallback|MockObject
+     */
+    private $subjectMassConsumerEnvelopeCallbackMock;
+
+    /**
+     * @var EnvelopeInterface|MockObject
+     */
+    private $messageMock;
+
+    /**
+     * @var Store|MockObject
+     */
+    private $storeMock;
+
+    protected function setUp(): void
+    {
+        $this->subjectMassConsumerEnvelopeCallbackMock = $this->createMock(SubjectMassConsumerEnvelopeCallback::class);
+        $this->storeMock = $this->createMock(Store::class);
+        $this->messageMock = $this->getMockForAbstractClass(EnvelopeInterface::class);
+
+        $this->envelopeFactoryMock = $this->createMock(EnvelopeFactory::class);
+        $this->storeManagerMock = $this->getMockForAbstractClass(StoreManagerInterface::class);
+        $this->loggerMock = $this->getMockForAbstractClass(\Psr\Log\LoggerInterface::class);
+
+        $objectManager = new ObjectManager($this);
+        $this->massConsumerEnvelopeCallbackPlugin = $objectManager->getObject(
+            MassConsumerEnvelopeCallback::class,
+            [
+                'envelopeFactory' => $this->envelopeFactoryMock,
+                'storeManager' => $this->storeManagerMock,
+                'logger' => $this->loggerMock,
+            ]
+        );
+    }
+
+    public function testAroundExecuteWhenApplicationHeadersDoesNotExist()
+    {
+        $this->messageMock->expects($this->once())
+            ->method('getProperties')
+            ->willReturn(null);
+
+        $isProceedCalled = false;
+        $proceed = function ($message) use (&$isProceedCalled) {
+            $isProceedCalled = !!$message;
```

Comment:
I don't think `!!$message` is a good option for marking if `$proceed` was called. Simply `true` is good enough.
```suggestion
            $isProceedCalled = true;
```

Reply:
Thank you for the suggestion.

In this case, if I use `$isProceedCalled = true;` then $message is not used anywhere and can be warned a coding convention. Also, I want to confirm that `$message` was put as a parameter on ` $proceed($message);`
Ak, maybe it is better if we change to `$isProceedCalled = (true === $message instead of EnvelopeInterface);`